src/main/kotlin/no/nav/familie/ef/iverksett/behandlingstatistikk/BehandlingstatistikkService.kt

```diff
@@ -7,14 +7,15 @@ import org.springframework.stereotype.Service
 import java.time.ZonedDateTime
 
 @Service
-class BehandlingstatistikkService(private val behandlingstatistikkRepository: BehandlingstatistikkRepository) {
+class BehandlingstatistikkService(private val behandlingstatistikkProducer: BehandlingstatistikkProducer,
+                                  private val behandlingstatistikkRepository: BehandlingstatistikkRepository) {
 
     fun lagreBehandlingstatistikk(behandlingstatistikk: BehandlingStatistikkDto) {
```

Comment:
Burde denne være `@Transactional`? Hvis det skjer en exception i kafka-steget, så vil statistikken ligge i DB som om den ble sendt. Fordeler og ulemper med det, selvsagt. Å lagre den åpner for rekjøring. Men antar en exception her resulterer i en serverside-feil for klienten, som ikke kan avgjøre om request'en ble lagret eller ikke, og dermed bør sende på nytt. Da oppstår det en duplikat. Alternativt kan man gå for et langt mer komplisert regime - lagre, og så kjøre en jobb (task) som sjekker hvilke som ikke er suksessfullt sendt til kafka, og oppdatere status i etterkant.

Reply:
Godt poeng, denne burde kanskje være transactional. Men det kan ikke oppstå duplikater akkurat nå (pga primærnøkkel)

Reply:
Men uten `@Transactional` risikerer vi å ha behandlinger i "limbo-tilstand" - de er lagret, men ikke oversendt til kafkakøen. Sleng på `Transactional` så slipper vi å tenke på den delen, kanskje?